torchci/pages/test/[[...page]].tsx

```diff
@@ -0,0 +1,278 @@
+  }
+  const chartSeries = formatSeries(chartData);
+
+  return (
+    <Grid item xs={12} lg={6} height={ROW_HEIGHT}>
+      <TimeSeriesPanelWithData
+        data={chartData}
+        series={chartSeries}
+        title={title}
+        yAxisLabel={yAxisLabel}
+        yAxisRenderer={(unit) => `${unit}`}
+      />
+    </Grid>
+  );
+}
+
+function GetUsage({
+  jobName,
+  workflowName,
+  testFile,
+  testClass,
+  workflowIds,
+  jobIds,
+}: {
+  jobName: string;
+  workflowName?: string;
+  testFile?: string;
+  testClass?: string;
+  workflowIds: string[];
+  jobIds: string[];
+}) {
+  const params = {
+    workflowName: workflowName,
+    jobName: jobName,
+    testFile: testFile,
+    testClass: testClass,
+    workflowIds: workflowIds,
+    jobIds: jobIds,
+  };
+
+  const url = `/api/usage-log-aggregator/lambda?params=${encodeURIComponent(
+    JSON.stringify(params)
+  )}`;
+  const { data } = useSWR(url, fetcher);
+
+  if (data === undefined || data.length == 0) {
+    return (<div></div>);
+  }
+
+  const transformedData = [];
+  // Transform the data a bit to fit into the same rockset schema
+  for (var idx in data["timestamp"]) {
+    transformedData.push({
+      timestamp: data["timestamp"][idx],
+      cpu: data["cpu"][idx],
+      mem: data["mem"][idx] / TO_GB,
+      gpu: data["gpu"][idx],
+      gpu_mem: data["gpu_mem"][idx] / TO_GB,
+    });
+  }
+
+  return (
+    <Grid container spacing={2}>
+      <DisplayInsights
+        data={transformedData}
+        title={"CPU usage (%)"}
+        timeFieldName={"timestamp"}
+        yAxisFieldName={"cpu"}
+      />
+
+      <DisplayInsights
+        data={transformedData}
+        title={"Memory usage"}
+        timeFieldName={"timestamp"}
+        yAxisFieldName={"mem"}
+        yAxisLabel={"GB"}
+      />
+
+      <DisplayInsights
+        data={transformedData}
+        title={"GPU usage (%)"}
+        timeFieldName={"timestamp"}
+        yAxisFieldName={"gpu"}
+      />
+
+      <DisplayInsights
+        data={transformedData}
+        title={"GPU memory usage"}
+        timeFieldName={"timestamp"}
+        yAxisFieldName={"gpu_mem"}
+        yAxisLabel={"GB"}
+      />
+    </Grid>
+  );
+}
+
+function GetJobs({
+  jobName,
+  workflowName,
+  testFile,
+  testClass,
+  workflowId,
+  jobId,
+  queryParams,
+}: {
+  jobName: string;
+  workflowName?: string;
+  testFile?: string;
+  testClass?: string;
+  workflowId?: string;
+  jobId?: string;
+  queryParams: RocksetParam[];
+}) {
+  const queryCollection = "commons";
+  const queryName = "test_insights_latest_runs";
+
+  const workflowIds = [];
+  const jobIds = [];
+
+  const url = `/api/query/${queryCollection}/${queryName}?parameters=${encodeURIComponent(
+    JSON.stringify(queryParams)
+  )}`;
+  const { data, error } = useSWR(url, fetcher);
+
+  // If there is no workflow and job ID specified, query Rockset for the list of N latest jobs
+  if (workflowId == null || jobId == null) {
```

Comment:
How do you get into this state? When I tried navigating to just `https://torchci-git-fork-huydhn-test-insights-hud-fbopensource.vercel.app/test/insights` I got an empty page

Reply:
Fixed. This should throw a proper warning about missing job name, which is a required parameter